umockdev: enable strictDeps

flokli opened the pull request:
As pointed out in
https://github.com/NixOS/nixpkgs/issues/280697#issuecomment-1906206490, umockdev tests shell out to udevadm, and in both cases, systemdMinimal should be sufficient.

Fixes #280697

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

flokli committed:
umockdev: enable strictDeps

As pointed out in
https://github.com/NixOS/nixpkgs/issues/280697#issuecomment-1906206490,
umockdev tests shell out to udevadm, and in both cases, systemdMinimal
should be sufficient.

Fixes #280697

jtojnar commented on pkgs/development/libraries/umockdev/default.nix:
```diff
@@ -59,10 +59,13 @@ stdenv.mkDerivation (finalAttrs: {
   nativeCheckInputs = [
     python3
-    which
+    systemdMinimal
```
Maybe we should patch it in https://github.com/martinpitt/umockdev/blob/6a1d57a80a511c8a6cc8a0756673d4f5f0f53146/src/umockdev-record.vala#L226 so that it works at runtime?

hcsch replied:
Ah, I completely forgot that when looking at https://github.com/NixOS/nixpkgs/issues/280697. The way it is now, `udevadm` isn't necessarily in the program's path and even if it is, it might not be from the same systemd version it has in its inputs. Patching that is kinda the only clean way, right?

flokli replied:
We either require everything using umockdev to provide a udevadm binary in the environment umockdev runs in, or we patch in some path to it.

Considering it's only used for testcases, either way is probably fine?

jtojnar replied:
I guess most of the packages tested with the help of umockdev already depend on systemd. But patching it here explicitly might be nicer in case there is an exception to this.

flokli replied:
I patched it in, and could then remove explicitly referencing to it from `nativeCheckInputs`. PTAL.

flokli committed:
umockev: hardcode udevadm

Replace references to udevadm with an absolute paths, so programs using
umockdev will just work without having to provide it in their test
environment $PATH.

This allows us removing systemdMinimal from nativeCheckInputs ourselves.
It's still needed in buildInputs, as the build queries for it via
pkg-config.
